src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using AutoRest.CSharp.AutoRest.Plugins;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Utilities;
+
+namespace AutoRest.CSharp.Mgmt.Decorator
+{
+    internal static class CodeModelExtension
+    {
+        private static readonly ConcurrentDictionary<string, string> _cache = new ConcurrentDictionary<string, string>();
```

Comment:
We already have problems with static dictionaries. Please don't add another one.

Reply:
Sure, I will introduce a singleton class for the renaming logic.
Out of curiosity, what problems are from the static dictionaries? One of them I know is that static dictionaries will consume a lot of memories.

Reply:
I change the cache to a local variable and pass it to every function in the naming transformation process.
We truly do not need to preserve this cache forever, therefore I believe a local variable will be enough